### PR TITLE
Fixed bug with rejecting correct value in spell checker. Fixed #121

### LINKED ISSUE #121: wrong suggestion when string requested on hash and keys are symbols
e.g.

```ruby
>> { aaa_a: 1, aaa_b: 1 }.fetch("aaa_b")
KeyError (key not found: "aaa_b")
Did you mean?  :aaa_a
```

ruby -v: `ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin18]`

verified with:

```ruby
DidYouMean::SpellChecker.new(dictionary: [:aaa_a, :aaa_b]).correct("aaa_a")
=> [:aaa_b]
```

### PR DESCRIPTION
**Pull request by makketagg**

### CODE REVIEW

**Comment by yuki24** [comment]
Thanks for the PR! Overall this looks great, and I think we can merge with a few minor changes.

File: lib/did_you_mean/spell_checkers/key_error_checker.rb [hunk 1]
```diff
@@ -8,7 +8,13 @@ def initialize(key_error)
     end
 
     def corrections
-      @corrections ||= SpellChecker.new(dictionary: @keys).correct(@key).map(&:inspect)
+      @corrections ||= similar_keys + SpellChecker.new(dictionary: @keys).correct(@key).map(&:inspect)
+    end
+
+    private
+
+    def similar_keys
```
Comment (yuki24) [inline lib/did_you_mean/spell_checkers/key_error_checker.rb]: I would call this method `exact_matches` as it describes what it does better.

File: lib/did_you_mean/spell_checkers/key_error_checker.rb [hunk 2]
```diff
@@ -8,7 +8,13 @@ def initialize(key_error)
     end
 
     def corrections
-      @corrections ||= SpellChecker.new(dictionary: @keys).correct(@key).map(&:inspect)
+      @corrections ||= similar_keys + SpellChecker.new(dictionary: @keys).correct(@key).map(&:inspect)
```
Comment (yuki24) [inline lib/did_you_mean/spell_checkers/key_error_checker.rb]: I think it makes more sense to only suggest exact matches if they are present, then fall back to the typical spell checking approach. That should by design prevent inaccurate suggestions from creeping into the final results.

```suggestion
      @corrections ||= exact_matches.empty? ? SpellChecker.new(dictionary: @keys).correct(@key).map(&:inspect) : exact_matches
```

Comment (makketagg) [reply lib/did_you_mean/spell_checkers/key_error_checker.rb]: Good point 👍

**Comment by yuki24** [comment]
Thanks!

**Pull request merged**